core/unit_test/TestNumericTraits.hpp

```diff
@@ -21,6 +21,19 @@
 #include <limits>
 #include "Kokkos_NumericTraits.hpp"
 
+// Suppress "'long double' is treated as 'double' in device code"
+#ifdef KOKKOS_COMPILER_NVCC
+#ifdef __NVCC_DIAG_PRAGMA_SUPPORT__
+#pragma nv_diagnostic push
+#pragma nv_diag_suppress 20208
+#else
+#ifdef __CUDA_ARCH__
+#pragma diagnostic push
+#pragma diag_suppress 20208
+#endif
+#endif
+#endif
```

Comment:
I feel that is not the right approach for that test.
We already have all sorts of guards for `long double` in these, I feel like we should consider reworking them like we did for the math function (I don;t remember the details but I see that it is not part of this PR meaning we somehow resolved it without suppressing the diagnostic altogether)

Reply:
We could just disable all tests for `long double` when we compile with `Cuda` support (which is what we are doing for other configurations) but these seem to work fine after suppressing the warning.

Reply:
Are the `long double` tests on CUDA already effectively switched off, in that if they are run as `double` in CUDA device code?

Reply:
Yes, everything that uses `TestNumericTraits` at least, see https://github.com/kokkos/kokkos/blob/63f05204d0eaeec83e25f87eeb8025a188683f92/core/unit_test/TestNumericTraits.hpp#L182-L204.